### PR TITLE
Drop unused machine fields

### PR DESCRIPTION
**Pull request by enxebre**
drop unused machine fields:
- machine.spec.versions
- machine.spec.configsource -- we don't use dynamic kubelet config
- machine.spec.providerSpec.machineClass -- we are not shipping the machine class resource at this time
- machine.status.versions -- we are not setting this yet
- machine.status.conditions -- we are not setting these node conditions up to machine objects yet

Follow ups:
- Revendor cluster-api-provider-aws
- Revendor installer
- Update mao CRDs

### CODE REVIEW

File: config/rbac/rbac_role.yaml [hunk 1]
```diff
@@ -89,7 +89,6 @@ rules:
   - machine.openshift.io
   resources:
   - machines
-  - machines/status
```
Comment (ingvagabund) [inline config/rbac/rbac_role.yaml]: This needs to go back

Comment (frobware) [reply config/rbac/rbac_role.yaml]: > This needs to go back

Can you elaborate on the why? TIA.

**Review by frobware — COMMENTED** [review]
LGTM but waiting on https://github.com/openshift/cluster-api/pull/34/files/e0a3702b5f8dd60c8d8a114747d1957132079f95#r279309452

File: config/rbac/rbac_role.yaml (hunk 1 above)
Comment (ingvagabund) [reply config/rbac/rbac_role.yaml]: Since aws actuator is updating machine status. And, this PR is about removing fields, not about changing RBAC rules.

Comment (enxebre) [reply config/rbac/rbac_role.yaml]: done!

**Pull request merged**